modules/tensor_mechanics/src/materials/ComputeAxisymmetric1DFiniteStrain.C

```diff
@@ -0,0 +1,81 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "ComputeAxisymmetric1DFiniteStrain.h"
+#include "Assembly.h"
+#include "FEProblem.h"
+#include "MooseMesh.h"
+
+template<>
+InputParameters validParams<ComputeAxisymmetric1DFiniteStrain>()
+{
+  InputParameters params = validParams<Compute1DFiniteStrain>();
+  params.addClassDescription("Compute a strain increment and rotation increment for finite strains in an axisymmetric 1D problem");
+  params.addCoupledVar("scalar_strain", "Scalar variable for axisymmetric 1D problem");
+  params.addCoupledVar("variable_strain", "Nonlinear variable for axisymmetric 1D problem");
```

Comment:
Will do this and I need to update all the 1D classes and tests in BISON using the new formulation.